src/components/settingsModal/settingsModal.tsx

```diff
@@ -1,21 +1,25 @@
 import React, { useEffect, useState } from 'react'
 import './styles.css'
-// import { invoke } from "@tauri-apps/api/tauri"
+import { invoke } from "@tauri-apps/api/tauri"
 
 interface ModalProps {
   show: boolean;
   onClose: () => void;
 }
 
+interface Permissions {
+  [key: string]: boolean;
+}
+
 const SettingsModal: React.FC<ModalProps> = ({ show, onClose }) => {
   if (!show) return null;
 
-  // TODO: get the users current permisions json
-  const [toggles, setToggles] = useState({
-    Location: false,
-    Clipboard: false,
-    Screenshot: false
-  });
+  const [toggles, setToggles] = useState<Permissions>({})
+  useEffect(() => {
+    invoke("get_permissions").then((permissions: any) => {
+      setToggles(permissions as Permissions)
+    })
+  }, [])
```

Comment:
This is probably being called multiple times, maybe we should wrap it in the hasRef like we do in App.tsx to prevent that?

Reply:
it is being ran redundantly. when adding the useRef fix to this one, there is no effect. when adding the useRef fix to the useEffect with the toggles as a dependency, it causes the permissions.json to be updated with an empty object, and then subsequent openings of the settings modal show no permissions or toggles. I think we might just have to leave it as is